internal/bminventory/inventory.go

```diff
@@ -2843,6 +2847,23 @@ func (b *bareMetalInventory) PostStepReply(ctx context.Context, params installer
 }
 
 func (b *bareMetalInventory) handleReplyError(params installer.PostStepReplyParams, ctx context.Context, log logrus.FieldLogger, h *models.Host, exitCode int64) error {
+	if exitCode == MediaDisconnected {
+		statusInfo := fmt.Sprintf("%s - %s", string(models.HostStageFailed),
+			"Cannot read from the media (ISO) - media was likely disconnected")
+
+		// Install command reports its status with a different API, directly from the assisted-installer.
+		// Just adding our diagnose to the existing error message.
+		if swag.StringValue(h.Status) == models.HostStatusError && h.StatusInfo != nil {
+			statusInfo = fmt.Sprintf("%s\n%s", statusInfo, *h.StatusInfo)
```

Comment:
Correct me if I'm wrong but won't this forever keep appending lines to the status info while steps keep failing?

Reply:
Consider the possibility of adding a dedicated boolean host field for media disconnection, and adding a host validation that makes sure that field is False. Combining strings like this is unexpected and error prone

Reply:
I'm not sure the host status info field is even shown prior to installation, but validations definitely are.

So validations would be good before installation.

But I believe we don't run validations after installation has started so... might need a hybrid solution here.

Or alternatively - the UI should simply be aware of this new host field and have dedicated UI elements for it

Reply:
It changes the status of the host to Failed which stop the installation and append this message only in case of failure.

Reply:
I can check if the message already contains the disconnection message just for case